module/Finna/src/Finna/OrganisationInfo/OrganisationInfo.php

```diff
@@ -79,1584 +74,190 @@ class OrganisationInfo implements
-            'contactInfo' => $contactInfoToResult ?? '',
-        ];
-        return $result;
+        return $orgInfo;
     }
 
     /**
-     * Date data handling function for museums
+     * Get location details
      *
-     * @param string $day  Weekday
-     * @param array  $json Data from museum api
+     * @param array   $sectors    Sectors if known, empty array otherwise
+     * @param string  $id         Parent organisation ID
+     * @param string  $locationId Location ID
+     * @param ?string $startDate  Start date (YYYY-MM-DD) of opening times (default is Monday of current week)
+     * @param ?string $endDate    End date (YYYY-MM-DD) of opening times (default is Sunday of current week)
      *
      * @return array
      */
-    protected function getMuseumDaySchedule($day, $json)
-    {
-        $today = date('d.n.');
-        $currentHour = date('H:i');
-        $return = [
-            'times' => [],
-            'closed' => false,
-            'openNow' => false,
-        ];
-        $dayShortcode = substr($day, 0, 3);
-        $dayDate = new \DateTime("$day this week");
-        $schedule = $json['opening_time'] ?? [];
-        if (
-            empty($schedule["{$dayShortcode}_start"])
-            || 'NULL' === $schedule["{$dayShortcode}_start"]
-            || empty($schedule["{$dayShortcode}_end"])
-            || 'NULL' === $schedule["{$dayShortcode}_end"]
-        ) {
-            $return['closed'] = true;
-        } else {
-            $time = [];
-
-            $time['opens'] = $this->formatTime($schedule["{$dayShortcode}_start"]);
-            $time['opens_datetime']
-                = $this->formatDateTime($dayDate, $schedule["{$dayShortcode}_start"]);
-            $time['closes'] = $this->formatTime($schedule["{$dayShortcode}_end"]);
-            $time['closes_datetime']
-                = $this->formatDateTime($dayDate, $schedule["{$dayShortcode}_end"]);
-            $return['times'][] = $time;
+    public function getDetails(
+        array $sectors,
+        string $id,
+        string $locationId,
+        ?string $startDate = null,
+        ?string $endDate = null
+    ): array {
+        if (!$sectors) {
+            $sectors = $this->getSectorsForOrganisation($id);
         }
-        $return['day'] = $this->translator->translate("day-name-short-$day");
-        $return['date'] = $dayDate->format('d.n.');
-        if ($today === $return['date']) {
-            $return['today'] = true;
-            if (
-                $currentHour >= $json['opening_time']["{$dayShortcode}_start"]
-                && $currentHour <= $json['opening_time']["{$dayShortcode}_end"]
-            ) {
-                $return['openNow'] = true;
-            }
+        $orgInfo = [];
+        if (in_array('lib', $sectors)) {
+            $orgInfo = $this->kirkanta->getDetails($this->getLanguage(), $id, $locationId, $startDate, $endDate);
+        }
+        if (in_array('mus', $sectors)) {
+            $orgInfo = array_merge(
```

Comment:
Aloin miettimään et onkohan tällä mergellä mitään vaikutusta mihinkään. Jos sil ei oo niin sit nää vois nätisti putkittaa one-linereilla esim return $this->getProvider(...)->getDetails(...)

Reply:
Muokattu. Jätin sen alunperin sinne, kun oli vanhassakin, mutta ei se varmaan oikeasti mitään järkevää tuota. Ei tuossa ole muutenkaan oikein järkeä, että välitetään monta sektoria, mutta nyt se olettaa, että on kirjasto, ellei ole museo mukana.